services/src/main/scala/cromwell/services/instrumentation/impl/statsd/StatsDConfig.scala

```diff
@@ -0,0 +1,33 @@
+package cromwell.services.instrumentation.impl.statsd
+
+import cats.data.Validated._
+import cats.syntax.cartesian._
+import cats.syntax.validated._
+import com.typesafe.config.Config
+import lenthall.exception.MessageAggregation
+import lenthall.validation.ErrorOr.ErrorOr
+import lenthall.validation.Validation._
+import net.ceedubs.ficus.Ficus._
+
+import scala.concurrent.duration.FiniteDuration
+
+case class StatsDConfig(hostname: String, port: Int, prefix: Option[String], flushRate: FiniteDuration)
+
+object StatsDConfig {
+  def apply(serviceConfig: Config) = {
+    val statsDConfig = serviceConfig.getConfig("statsd")
+
+    val hostname: ErrorOr[String] = validate[String] { statsDConfig.as[String]("hostname") }
+    val port: ErrorOr[Int] = validate[Int] { statsDConfig.as[Int]("port") }
+    val prefix: ErrorOr[Option[String]] = statsDConfig.as[Option[String]]("prefix").validNel
+    val flushRate: ErrorOr[FiniteDuration] = statsDConfig.as[FiniteDuration]("flush-rate").validNel
+
+    (hostname |@| port |@| prefix |@| flushRate).tupled match {
+      case Valid((h, p, n, f)) => new StatsDConfig(h, p, n, f)
+      case Invalid(message) => throw new IllegalArgumentException with MessageAggregation {
+        override val exceptionContext = "StatsD config is invalid"
+        override val errorMessages = message.toList
+      }
+    }
```

Comment:
Yeah I thought the same thing when writing this: `You're about to get catted`

Reply:
What's the ultimate outcome of throwing here? If it's that the system shuts down before Cromwell gets started then seems ok to me. If it's that stats mysteriously disappear, not so much...

Otherwise, the `|@|` syntax is deprecated by cats 1.0.0, you'll want to `(hostname, port, flushRate) mapN { ... }`

Reply:
When a service fails to start it *should* be shutting down Cromwell. That's been working for me over in health monitor land.

Reply:
Yes the desired outcome is for the actor not to start properly which should be caught by the supervisor and prevent Cromwell from starting.

Reply:
Great! In that case throwing seems appropriate to me